test/csuite/wt7989_compact_checkpoint/main.c

```diff
@@ -277,32 +259,26 @@ thread_func_checkpoint(void *arg)
     struct thread_data *td;
     WT_RAND_STATE rnd;
     WT_SESSION *session;
-    uint64_t sleep_sec;
+    uint64_t ready_counter_local, sleep_sec;
     int i;
-    bool signalled;
 
     td = (struct thread_data *)arg;
 
     testutil_check(td->conn->open_session(td->conn, NULL, NULL, &session));
 
     __wt_random_init_seed((WT_SESSION_IMPL *)session, &rnd);
 
-    if (td->cond != NULL) {
-        /*
-         * Spawn the compact thread here to make sure the both threads are ready for the synced
-         * start.
-         */
-        testutil_check(pthread_create(&thread_compact, NULL, thread_func_compact, td));
-
-        printf("Waiting for the signal...\n");
-        /*
-         * Wait for the signal and time out after 20 seconds. wait_run_check is required because the
-         * time out is longer that one second.
-         */
-        __wt_cond_wait_signal(
-          (WT_SESSION_IMPL *)session, td->cond, 20 * WT_MILLION, wait_run_check, &signalled);
-        testutil_assert(signalled);
-        printf("Signal received!\n");
+    if (!td->stress_test) {
```

Comment:
This is code duplication. Let's try and extract it into a separate method. We can print "Threads ready..." messages in the caller.

Reply:
Fixed.